src/rest/api.go

```diff
@@ -346,6 +353,67 @@ func GetConfiguration(w http.ResponseWriter, r *http.Request) {
 	w.Write(resp)
 }
 
+func AddFileToZip(zw *zip.Writer, filename, iniFilePath string) bool {
+	fh := &zip.FileHeader{
+		Name:   filename,
+		Method: zip.Deflate,
+	}
+	fh.SetModTime(time.Now())
+
+	fw, err := zw.CreateHeader(fh)
+	if err != nil {
+		log.Printf("Error creating zip header: %v", err)
+		return false
+	}
+
+	dat, err := ioutil.ReadFile(iniFilePath)
+	if err != nil {
+		log.Printf("Error reading file: %v", err)
+		return false
+	}
+	
+	_, err = fw.Write(dat)
+	if err != nil {
+		log.Printf("Error writing %s config to zip: %v", filename, err)
+		return false
+	}
+
+	return true
+}
+
+func downloadConfiguration(w http.ResponseWriter, r *http.Request) {
```

Comment:
Rename to downloadConfigurationHandler and outsource logic (except for the part that does the parsing and rewriting header) into a new file inside the configuration package. AddFileToZip should be outsourced as well. The rest package should stay as small as possible.